.github/workflows/pr.yml

```diff
@@ -0,0 +1,43 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: build
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.6
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+      - name: Setup Dev Environment
+        run: |
+          pip install virtualenv
+          virtualenv .venv
+          source .venv/bin/activate
+          ./run setup
+      - name: Check formatting
+        run: |
+          source .venv/bin/activate
+          ./run check -f
+      - name: Run Unit tests
+        run: |
+          source .venv/bin/activate
+          ./run test all unit
+      - name: Coverage
+        run: |
+          source .venv/bin/activate
+          coverage xml
+      - name: Upload coverage
+        uses: codecov/codecov-action@v1
```

Comment:
Please update this to v2 as we need to do anyway early next year when v1 is no longer available.

https://github.com/codecov/codecov-action